clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -289,41 +288,191 @@ public void testAppend() throws IOException {
         fs.append(null, 0, null);
     }
 
-    /*
+    private void mockNonExistingPath(ObjectLocation objectLoc) throws ApiException {
+        when(objectsApi.statObject(objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath()))
+                .thenThrow(new ApiException(HttpStatus.SC_NOT_FOUND, "no such file"));
+        when(objectsApi.listObjects(objectLoc.getRepository(), objectLoc.getRef(),
+                objectLoc.getPath() + Constants.SEPARATOR, "", 1, ""))
```

Comment:
Also I don't understand the trailing `Constants.SEPARATOR`: if you specify that `a/b/c` does not exist, I would expect that listing `a/b/c` would _also_ not exist.